urllib3/util/response.py

```diff
@@ -20,3 +25,33 @@ def is_fp_closed(obj):
         pass
 
     raise ValueError("Unable to determine whether fp is closed.")
+
+
+def validate_headers(headers):
+    """
+    Checks whether headers have been successfully and completely parsed.
+
+    Only works on Python 3.
+
+    :param headers: Headers to check.
+    :type headers: `httplib.HTTPMessage`.
+
+    :returns: Whether any errors occured during header parsing.
+    :rtype: bool
```

Comment:
Generally I'd be in favour of going with YAGNI here and abandoning it. However, if you decide to keep it, I'd invert the logic: a method called `validate_headers` should probably return `True` if the headers are valid and `False` if they aren't.